### PR TITLE
EN-64862: add hotfix workflow

### PR DESCRIPTION
**Pull request by sarah-salinas**
This commit adds a hotfix workflow to the build process.

Compare with current hotfix workflows in [core](https://github.com/socrata/core/blob/main/Jenkinsfile), [frontend](https://github.com/socrata/platform-ui/blob/main/frontend/Jenkinsfile) and [storyteller](https://github.com/socrata/platform-ui/blob/main/storyteller/Jenkinsfile).

### CODE REVIEW

**Comment by sarah-salinas** [comment]
## Testing

### Simulated Main Build

https://jenkins-build.socrata.com/job/soda-fountain-branch/24/console

- Hotfix
- Build
- Docker Build
- Publish Image: internal only
- Deploy: staging

Description: 

### Release Dry Run


### Simulated Release Build


### Simulated Hotfix

https://jenkins-build.socrata.com/job/data-platform/job/soda-fountain/job/A23-HF/2/console - 🟢 

Stages:

- Hotfix
- Build
- Docker Build, Post: pushed hotfix tag to repo
- Publish Image: all environments, Post: rms create build
- Deploy: rc, Post: rms create deployment

Description: `soda-fountain:A23-HF-0`

File: Jenkinsfile [hunk 1]
```diff
@@ -80,23 +102,25 @@ pipeline {
...
       steps {
         script {
           lastStage = env.STAGE_NAME
-          if (params.RELEASE_BUILD) {
+          if (params.RELEASE_BUILD || isHotfix) {
+            env.VERSION = (isHotfix) ? env.HOTFIX_NAME : params.RELEASE_NAME
               env.DOCKER_TAG = dockerize.dockerBuildWithSpecificTag(
               tag: params.RELEASE_NAME,
-              path: sbtbuild.getDockerPath(),
+              path: env.DOCKER_PATH,
```
Comment (sarah-salinas) [inline Jenkinsfile:116]: While intending to test out whether we could use sbtbuild.getDockerPath(), I accidentally pushed the change to main:  https://github.com/socrata-platform/soda-fountain/commit/616dc94a3cf1b2597475ac931fc0c03c3d10171b

Comment (sarah-salinas) [reply Jenkinsfile:116]: One test after fixing this issue:  https://jenkins-build.socrata.com/job/soda-fountain-branch/31/console

**Review by rjmac — APPROVED** [review]

**Comment by sarah-salinas** [comment]
Sanity check after PR feedback updates:  https://jenkins-build.socrata.com/job/soda-fountain-branch/32/console - 🟢

**Review by ds2643 — APPROVED** [review]
These changes reflect those made on other repositories.

**Pull request merged**